core/src/main/java/hudson/Util.java

```diff
@@ -1415,18 +1422,36 @@ public static Number tryParseNumber(@CheckForNull String numberStr, @CheckForNul
     }
 
     /**
-     * Checks if the public method defined on the base type with the given arguments
-     * are overridden in the given derived type.
+     * Checks if the method defined on the base type with the given arguments
+     * is overridden in the given derived type.
      */
     public static boolean isOverridden(@Nonnull Class base, @Nonnull Class derived, @Nonnull String methodName, @Nonnull Class... types) {
         try {
-            return !base.getMethod(methodName, types).equals(
-                    derived.getMethod(methodName,types));
+            return !getMethod(base, methodName, types).equals(getMethod(derived, methodName, types));
         } catch (NoSuchMethodException e) {
             throw new AssertionError(e);
         }
     }
 
+    private static Method getMethod(@Nonnull Class clazz, @Nonnull String methodName, @Nonnull Class... types) throws NoSuchMethodException {
```

Comment:
Sorry, I've missed the obvious code block :(
From my POW, it would be better not to throw runtime exceptions. Null might be better here, because the calling method should be able the case. 

Currently we have a runtime exception, which may be thrown away from the new isOverriden() method and which has no documentation or such runtime exception. This behavior is very hazardous for a widely used method.